frameworks/flaml/exec.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import warnings
+import pandas as pd
+
+from amlb.utils import Timer
+
+from flaml import AutoML, __version__
+
+from frameworks.shared.callee import call_run, result, save_metadata
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
+    log.info("\n**** FLAML ****\n")
+    save_metadata(config, version=__version__)
+    time_budget = config.max_runtime_seconds
+    n_jobs = config.framework_params.get('_n_jobs', config.cores)
+
+    print("Running FLAML with {} number of cores".format(config.cores))
+
+    is_classification = config.type == 'classification'
+    column_names, _ = zip(*dataset.columns)
+    column_types = dict(dataset.columns)
+    train = pd.DataFrame(dataset.train.data, columns=column_names).astype(
+        column_types, copy=False)
+    label = dataset.target.name
+    X_train = y_train = None
+    test = pd.DataFrame(dataset.test.data, columns=column_names).astype(
+        column_types, copy=False)
+    X_test = test.drop(columns=label)
+    y_test = test[label]
+
+    aml = AutoML()
+
+    # Mapping of benchmark metrics to flaml metrics
+    metrics_mapping = dict(
+        acc='accuracy',
+        auc='roc_auc',
+        f1='f1',
+        logloss='log_loss',
+        mae='mae',
+        mse='mse',
+        rmse='rmse',
+        r2='r2',
+    )
+
+    metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if metric is None:
+        raise ValueError("Performance metric {} not supported.".format(config.metric))
+
+    perf_metric = metrics_mapping[
+        config.metric] if config.metric in metrics_mapping else 'auto'
+    if perf_metric is None:
+        log.warning("Performance metric %s not supported.", config.metric)
+
+    training_params = {k: v for k, v in config.framework_params.items()
+     if not k.startswith('_')}
+
+    with Timer() as training:
+        aml.fit(X_train, y_train, train, label, perf_metric, config.type, 
+            n_jobs=n_jobs, 
+            log_file_name=config.output_predictions_file.replace('.csv','.log'),
```

Comment:
we're trying to put logs in a dedicated folder:
```python
from frameworks.shared.callee import  …,  output_subdir
…
log_dir = output_subdir("logs", config)  # this creates a subfolder unique for the given task/fold
…
log_file_name=os.path.join(log_dir, "flaml.log")
```

Reply:
Thanks for the info. I have revised accordingly.